cms/djangoapps/contentstore/models.py

```diff
@@ -308,13 +346,49 @@ def filter_links(
     ) -> QuerySet["EntityLinkBase"]:
         """
         Get all links along with sync flag, upstream context title and version, with optional filtering.
+
+        `use_top_level_parents` is an special filter, replace any result with the top-level parent if exists.
+        Example: We have linkA and linkB with top-level parent as linkC and linkD without top-level parent.
+        After all other filters:
+        Case 1: `use_top_level_parents` is False, the result is [linkA, linkB, linkD]
+        Case 2: `use_top_level_parents` is True, the result is [linkC, linkD]
         """
-        ready_to_sync = link_filter.pop('ready_to_sync', None)
-        result = cls.objects.filter(**link_filter).select_related(
+        RELATED_FIELDS = [
             "upstream_container__publishable_entity__published__version",
             "upstream_container__publishable_entity__learning_package",
             "upstream_container__publishable_entity__published__publish_log_record__publish_log",
-        ).annotate(
+        ]
+
+        ready_to_sync = link_filter.pop('ready_to_sync', None)
+        use_top_level_parents = link_filter.pop('use_top_level_parents', None)
```

Comment:
Same minor comment here about making this an explicit separate parameter instead of putting it into `link_filter`.

Reply:
Updated https://github.com/openedx/edx-platform/pull/37076/commits/e77a67f214681adbbb09a5a58e07b8a1c2800e57